### ISSUE TITLE
Mouse recording is broken

### ISSUE THREAD

**Issue by L1nkus** [RMPR/atbswp#16]
![atbswp-2020-05-24_13 07 24](https://user-images.githubusercontent.com/12627588/82752710-3e233d80-9dc0-11ea-90d3-52e5935b72ac.gif)

**Comment by Nivekiba** [comment]
Show the output of your recording file, in `/tmp/atbswp` or where you saved it

**Comment by RMPR** [comment]
Please, provide more details: OS? WM? Python version?

**Comment by L1nkus** [comment]
[atbswp_recent.txt](https://github.com/RMPR/atbswp/files/4674543/atbswp_recent.txt)

Had to re-record it, but it's a similar movement and result. Also, 1920x1080 vs 1024x768 that was there originally.
Python 3.8.2
Linux 5.6.8 (Arch)
i3wm 4.13-158-gf0e6dcbf (gaps-next)
Xorg 1.20.8

**Comment by RMPR** [comment]
Can reproduce the bug with :
Python 3.7
Linux 5.3 (Fedora 31)
KDE 5
Xorg 1.20.6

It seems like a bug with pynput, fwiw the clicks works normally, the bug only affects the mouse movement.

**Comment by L1nkus** [comment]
When I tried it in windows 10, it behaves similarly.

![out](https://user-images.githubusercontent.com/12627588/82846794-e8977f80-9eea-11ea-97a7-84b72208852f.gif)


[atbswp-20200525.txt](https://github.com/RMPR/atbswp/files/4679102/atbswp-20200525.txt)


Also, when moving the mouse into 0 x or y coordinate, on windows, the program crashes:

```
Unhandled exception in listener callback
Traceback (most recent call last):
  File "C:\Users\autor\AppData\Local\Programs\Python\Python38\lib\site-packages\pynput\_util\win32.py", line 382, in _handler
    converted = self._convert(code, msg, lpdata)
  File "C:\Users\autor\AppData\Local\Programs\Python\Python38\lib\site-packages\pynput\_util\win32.py", line 397, in _convert
    raise NotImplementedError()
NotImplementedError

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\autor\AppData\Local\Programs\Python\Python38\lib\site-packages\pynput\_util\__init__.py", line 162, in inner
    return f(self, *args, **kwargs)
  File "C:\Users\autor\AppData\Local\Programs\Python\Python38\lib\site-packages\pynput\_util\win32.py", line 386, in _handler
    self._handle(code, msg, lpdata)
  File "C:\Users\autor\AppData\Local\Programs\Python\Python38\lib\site-packages\pynput\mouse\_win32.py", line 186, in _handle
    self.on_move(data.pt.x, data.pt.y)
  File "C:\Users\autor\AppData\Local\Programs\Python\Python38\lib\site-packages\pynput\_util\__init__.py", line 78, in inner
    if f(*args) is False:
  File "C:\Users\autor\Downloads\atbswp-master\atbswp\control.py", line 148, in on_move
    self.write_mouse_action(move="moveTo", parameters=f"{x}, {y}")
  File "C:\Users\autor\Downloads\atbswp-master\atbswp\control.py", line 118, in write_mouse_action
    and coordinates[1] - self._lasty < self.mouse_sensibility:
IndexError: list index out of range
```

Windows 10
Python 3.8.0

**Comment by RMPR** [comment]
> Unhandled exception in listener callback
> Traceback (most recent call last):
>   File "C:\Users\autor\AppData\Local\Programs\Python\Python38\lib\site-packages\pynput\_util\win32.py", line 382, in _handler
>    converted = self._convert(code, msg, lpdata)
>  File "C:\Users\autor\AppData\Local\Programs\Python\Python38\lib\site-packages\pynput\_util\win32.py", line 397, in _convert
>    raise NotImplementedError()
> NotImplementedError

Definitely a bug with pynput

> Also, when moving the mouse into 0 x or y coordinate, on windows, the program crashes:

This can be solved easily by handling the exception properly.

**Comment by L1nkus** [comment]
So, on which side is the issue with mouse movement?
If I run a pynput script (on the linux setup), the output seems fine actually, unless the info is parsed from a different form.

[pynputrec.txt](https://github.com/RMPR/atbswp/files/4683656/pynputrec.txt)

**Comment by RMPR** [comment]
Everything happens in the control.py, specifically, the functions for the mouse are `on_move`, `on_click`, `on_scroll`, obviously what you're describing happens within `on_move`, if you want to solve this issue by yourself, I can provide some guidance.

**Comment by RMPR** [comment]
Just pushed a fix for the crashes on Windows when the mouse reaches x=0 or y=0, can you confirm this is solving your problem? https://github.com/RMPR/atbswp/commit/9534bcf0a9d3a42c11f79b1b1f7b0bc5f3af8494

**Comment by L1nkus** [comment]
Since on windows, the parameter string sometimes contains a negative coordinate in place of a 0, it was then parsed into an incomplete list, and further uncaught exceptions related to size != 2 occurred later on.
Getting rid of isdigit(), which doesn't work on negative numbers, solves that.
#17

**Comment by JohnParten** [comment]
Hello,
I think I have also encountered this bug and was hoping it would get fixed.
My setup is ...
Python 3.7.7
Linux 5.6.19-200.fc31.x86_64
KDE Plasma 5.17.5
XOrg 1.20.6

I tried to use KolourPaint as a test. I drew 4 horizontal lines and 1 vertical line. I think each line should have been ...
a) Mouse moveTo (for starting point of this line)
b) Mouse left button down
c) Mouse moveTo (for drawing the line)
d) Mouse left button up

The log file (just 23 lines)  is attached to this comment but it does not match what I was expecting so any guidance appreciated ~  Happy to do more tests if it helps to fix issue ~ JohnP
[nomouse.txt](https://github.com/RMPR/atbswp/files/4842724/nomouse.txt)
